### ISSUE TITLE
UnhandledPromiseRejectionWarning: Error: Could not retrieve ... (500: unknown error)

### ISSUE THREAD

**Issue by ddeboer** [comunica/comunica#777]
#### Issue type:

- :bug: Bug 

____
#### Description:

When querying URLs that throw a 500 (for instance #775, because of a misconfigured server), Comunica doesn’t handle the error. 

We would expect this snippet to print `error!`:

```javascript
 try {
      await newEngine().query(
        `SELECT * WHERE { ?s ?p ?o }`,
        {sources: ['https://data.bibliotheken.nl/doc/dataset/rise-alba']}  // URL that responds with 500.
      );
    } catch (e) {
      console.log('error!');
    }
```

But it throws the crash log pasted below instead.

____
#### Environment:

```
| software            | version
| ------------------- | -------
| Comunica Init Actor | 1.19.0
| node                | v14.15.2
| npm                 | 7.4.0
| yarn                | 1.22.10
| Operating System    | darwin (Darwin 20.2.0)
```

#### Crash log:

```
(node:57032) UnhandledPromiseRejectionWarning: Error: Could not retrieve https://data.bibliotheken.nl/doc/dataset/rise-alba (500: unknown error)
    at ActorRdfDereferenceHttpParse.run (/Users/david/src/netwerk-digitaal-erfgoed/registry-api/node_modules/@comunica/actor-rdf-dereference-http-parse/lib/ActorRdfDereferenceHttpParseBase.js:56:27)
    at MediatedLinkedRdfSourcesAsyncRdfIterator.getSource (/Users/david/src/netwerk-digitaal-erfgoed/registry-api/node_modules/@comunica/actor-rdf-resolve-quad-pattern-hypermedia/lib/MediatedLinkedRdfSourcesAsyncRdfIterator.js:49:38)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:57032) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 15)
(node:57032) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by rubensworks** [comment]
This behaviour is in fact intended.

The `query` method returns an object that contains result streams.
The query execution process only starts once the user starts consuming one of these streams.
This method will only throw if some illegal parameters were passed, but not for anything that goes wrong during query execution.

If you want to catch the errors during query execution, you'lll need to attach an `'error'` event handler to one of the result streams.
For example:
```javascript
const result = await myEngine.query(`
  SELECT ?s ?p ?o WHERE {
    ?s ?p <http://dbpedia.org/resource/Belgium>.
    ?s ?p ?o
  } LIMIT 100`, {
  sources: ['http://fragments.dbpedia.org/2015/en'],
});
...
result.bindingsStream.on('error', (error) => {
    console.error(error);
});
```

More information on this can be found here: https://comunica.dev/docs/query/getting_started/query_app/#3-1-consuming-binding-results-as-a-stream

**Comment by ddeboer** [comment]
I tried that too, but it yields the same behaviour with the URL in question:

```typescript
const result = await newEngine().query(
  `SELECT *  WHERE { ?s ?p ?o } LIMIT 100`, {
    sources: ['https://data.bibliotheken.nl/doc/dataset/rise-alba'],
  }
) as IQueryResultBindings;
result.bindingsStream.on('error', (error) => {
  // Never reached.
  console.error('err', error);
});

// (node:58018) UnhandledPromiseRejectionWarning: Error: Could not retrieve https://data.bibliotheken.nl/doc/dataset/rise-alba (500: unknown error)
```

**Comment by rubensworks** [comment]
Hmm, that _should_ catch the error.
I thought this problem was fixed in #565, so this is either a regression, or a case we missed.
